qemu/tests/boot_nbdimage_with_qsd.py

```diff
@@ -0,0 +1,55 @@
+from virttest import data_dir
+from virttest import qemu_storage
+
+from avocado import fail_on
+from avocado.utils import process
+from provider import qemu_img_utils as img_utils
+from provider.qsd import QsdDaemonDev
+
+
+def run(test, params, env):
+    """
+    Boot nbd image with qsd plus unix.
+
+    1) Prepare an installed OS image via converting
+    2) Export the OS image via QSD plus unix
+    3) Start VM from the exported image
+    """
+    def pre_test():
+        # prepare an installed OS image
+        test.log.info(
+            "Prepare an installed OS image file via converting, "
+            "and exporting it via QSD")
+        source = params["convert_source"]
+        target = params["convert_target"]
+        source_params = params.object_params(source)
+        root_dir = data_dir.get_data_dir()
+        source_image = qemu_storage.QemuImg(source_params, root_dir, source)
+
+        test.log.info("Convert %s to %s", source, target)
+        fail_on((process.CmdError,))(source_image.convert)(params, root_dir)
+
+        # export the OS image over QSD
+        qsd.start_daemon()
+
+    def run_test():
+        # boot up a guest from the exported OS image
+        test.log.info("Boot up a guest from the exported OS image.")
+        login_timeout = params.get_numeric("login_timeout", 360)
+        vm = None
```

Comment:
yes, we can remove vm at line 42

line 39, it's meaningful to set vm to None in case we hit exception in boot_vm_with_images, then vm is not defined, in this situation, at line 45, we need a check:
if vm:
       vm.destroy()

Reply:
Got it, thank you Zhenchao for your explanation.